pkg/controller-gen/main.go

```diff
@@ -61,6 +63,15 @@ func Run(opts cgargs.Options) {
 	clientGen := generators.NewClientGenerator()
 
 	getTargets := func(context *generator.Context) []generator.Target {
+		// replace the default formatter options to ensure unused imports are pruned.
+		// ref: https://github.com/kubernetes/gengo/pull/277#issuecomment-2557462569
+		if fileType, ok := context.FileTypes["go"]; ok {
+			if defaultFileType, ok := fileType.(*generator.DefaultFileType); ok {
+				defaultFileType.Format = func(src []byte) ([]byte, error) {
+					return imports.Process("", src, nil)
+				}
+			}
+		}
```

Comment:
I think that'd be safer. He's also mentioned using a `sync.Once` which I think we should, though I haven't digged into the gengo code to see why that would require it.

```suggestion
               goGenerator := generators.NewGoFile()
               goGenerator.Format = func(src []byte) ([]byte, error) {
					return imports.Process("", src, nil)
				}
				context.FileType[generators.GoFileType] = goGenerator
```

Reply:
We could wrap it in a sync.once but I added debug printfs to both this getTargets func and the custom Format function and it appears the context is reused through all the rest of the Execute calls so it's not necessary to replace it more than once. It also seems harmless TO replace it more than once though.

Reply:
I've confirmed that this also works for k3s

Reply:
Great. What do you think of using the suggested approach instead of doing a type check of an interface? That way this won't silently break when we update gengo.

Basically, instead of getting the default file generator for "go" filetype, let's just create it ourselves and modify its `Format` field.

Also, `"go"` -> generators.GoFileType.

Reply:
changed